include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -698,45 +719,114 @@ class graph_oriented_layout_design_impl
         {
             timeout = 10000u;
         }
-        bool timeout_limit_reached = false;
 
-        // main A* loop
+        // main loop
         while (!timeout_limit_reached)
         {
-            for (auto& ssg : ssg_vec)
+            // if multithreading is enabled
+            if (ps.enable_multithreading)
             {
-                if (ssg.frontier_flag)
+                // mutex to protect shared resources (if necessary)
```

Comment:
That's what a mutex always does :D Could you indicate what you are protecting specifically?